pkg/network/ebpf/c/protocols/tls/java/erpc.h

```diff
@@ -0,0 +1,216 @@
+    conn_tuple_t connection = {0};
+    const bool val = true;
+    u32 bytes_read = 0;
+
+    //interactive pointer to read the data buffer
+    void* bufferPtr = data;
+
+    //read the connection tuple from the ioctl buffer
+    if (0 != bpf_probe_read_user(&connection, sizeof(conn_tuple_t), bufferPtr)){
+        log_debug("[handle_sync_payload] failed to parse connection info\n");
+        return 1;
+    }
+    normalize_tuple(&connection);
+    bufferPtr+=sizeof(conn_tuple_t);
+
+    // read the actual length of the message (limited by HTTP_BUFFER_SIZE)
+    if (0 != bpf_probe_read_user(&bytes_read, sizeof(bytes_read), bufferPtr)){
+#ifdef DEBUG
+        u64 pid_tgid = bpf_get_current_pid_tgid();
+        u64 pid = pid_tgid >> 32;
+        log_debug("[handle_sync_payload] failed reading message length location for pid %d\n", pid);
+#endif
+        return 1;
+    }
+    bufferPtr+=sizeof(bytes_read);
+
+    // register the connection in our map
+    bpf_map_update_with_telemetry(java_tls_connections, &connection, &val, BPF_ANY);
+    log_debug("[handle_sync_payload] handling tls request of size: %d for connection src addr: %llx; dst address %llx\n",
+                bytes_read, connection.saddr_l, connection.daddr_l);
+    https_process(&connection, bufferPtr, bytes_read, JAVA_TLS);
+    http_batch_flush(ctx);
+    return 0;
+}
+
+static int __always_inline handle_close_connection(void *data) {
+
+    //read the connection tuple from the ioctl buffer
+    conn_tuple_t connection = {0};
+    if (0 != bpf_probe_read_user(&connection, sizeof(conn_tuple_t), data)){
+        log_debug("[java_tls_handle_close] failed to parse connection info\n");
+        return 1;
+    }
+    normalize_tuple(&connection);
+
+    void *exists = bpf_map_lookup_elem(&java_tls_connections, &connection);
+    // if the connection exists in our map, finalize it and remove from the map
+    // otherwise just ignore
+    if (exists != NULL){
+        https_finish(&connection);
+        bpf_map_delete_elem(&java_tls_connections, &connection);
+    }
+    return 0;
+}
+
+static int __always_inline handle_connection_by_peer(void *data) {
+
+    connection_by_peer_key_t peer_key ={0};
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    peer_key.pid = pid_tgid >> 32;
+
+    //interactive pointer to read the data buffer
+    void* bufferPtr = data;
+
+    //read the connection tuple from the ioctl buffer
+    conn_tuple_t connection = {0};
+    if (0 != bpf_probe_read_user(&connection, sizeof(conn_tuple_t), bufferPtr)){
+        log_debug("[handle_connection_by_peer] failed to parse connection info for pid: %d\n", peer_key.pid);
+        return 1;
+    }
+    normalize_tuple(&connection);
+    bufferPtr+=sizeof(conn_tuple_t);
+
+    //read the host tuple (peer domain string and peer port)
+    if (0 != bpf_probe_read_user(&peer_key.peer, sizeof(peer_t), bufferPtr)){
+        log_debug("[handle_connection_by_peer] failed reading hostname location for pid %d\n", peer_key.pid);
```

Comment:
i'm not sure hostname location is the best description to the code, did I miss anything?